### PR TITLE
Move connectedk8s CLI to track2 python SDK models

### PR DESCRIPTION
**Pull request by akashkeshari**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
connectedk8s migration to track 2

File: src/connectedk8s/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+1.1.6
+++++++
+* Moved to track2 SDK
+* `az connectedk8s connect`: Added onboarding timeout parameter
```
Comment (alphaWizard) [inline src/connectedk8s/HISTORY.rst:9]: Should we add _the release namespace thing_ we did few weeks back, here? [PR](https://github.com/Azure/azure-cli-extensions/pull/3611/files)

File: src/connectedk8s/azext_connectedk8s/_help.py [hunk 2]
```diff
@@ -26,6 +26,8 @@
       text: az connectedk8s connect -g resourceGroupName -n connectedClusterName --proxy-https https://proxy-url --proxy-http http://proxy-url --proxy-skip-range excludedIP,excludedCIDR,exampleCIDRfollowed,10.0.0.0/24
     - name: Onboard a connected kubernetes cluster by specifying the https proxy, http proxy, no proxy  with cert settings.
       text: az connectedk8s connect -g resourceGroupName -n connectedClusterName --proxy-cert /path/to/crt --proxy-https https://proxy-url --proxy-http http://proxy-url --proxy-skip-range excludedIP,excludedCIDR,exampleCIDRfollowed,10.0.0.0/24
+    - name: Onboard a connected kubernetes cluster with custom onboarding timeout.
+      text: az connectedk8s connect -g resourceGroupName -n connectedClusterName --onboarding-timeout 10
```
Comment (alphaWizard) [inline src/connectedk8s/azext_connectedk8s/_help.py:30]: We decided on parameter introduce for this?

Comment (alphaWizard) [reply src/connectedk8s/azext_connectedk8s/_help.py:30]: If so, should we add it for other commands(update_agents, enable_features, disable_features) which invokes _helm upgrade_ command? For consistency, imo.

File: src/connectedk8s/azext_connectedk8s/_params.py [hunk 3]
```diff
@@ -34,6 +34,7 @@ def load_arguments(self, _):
         c.argument('infrastructure', options_list=['--infrastructure'], help='The infrastructure on which the Kubernetes cluster represented by this connected cluster will be running on.', arg_type=get_enum_type(Infrastructure_Enum_Values))
         c.argument('disable_auto_upgrade', options_list=['--disable-auto-upgrade'], action='store_true', help='Flag to disable auto upgrade of arc agents.')
         c.argument('cl_oid', options_list=['--custom-locations-oid'], help="OID of 'custom-locations' app")
+        c.argument('onboarding_timeout', options_list=['--onboarding-timeout'], help='Time required (in minutes) for the arc-agent pods to be installed on the kubernetes cluster. Override this value if the hardware/network constraints on your cluster requires more time for installing the arc-agent pods.')
```
Comment (alphaWizard) [inline src/connectedk8s/azext_connectedk8s/_params.py:37]: seconds would be more standard time parameter?

Comment (alphaWizard) [reply src/connectedk8s/azext_connectedk8s/_params.py:37]: Should we group **"no_wait**" and **"onboarding_timeout"** parameters using _arg_group_? as they are both related to the wait time for onboarding of agent pods.

**Comment by jsntcy** [comment]
@akashkeshari, please re-record the test to make CI pass.

File: src/connectedk8s/HISTORY.rst (hunk 1 above)
Comment (akashkeshari) [reply src/connectedk8s/HISTORY.rst:9]: Added

File: src/connectedk8s/azext_connectedk8s/_help.py (hunk 2 above)
Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/_help.py]: The rest of the commands do not require an image pull operation(its just config update). The timeout was occuring for ASE due to delay in image pull & container creation

File: src/connectedk8s/azext_connectedk8s/_params.py (hunk 3 above)
Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/_params.py]: updated to seconds. added arg grp

**Review by alphaWizard — APPROVED** [review]

File: src/connectedk8s/azext_connectedk8s/_utils.py [hunk 4]
```diff
@@ -286,7 +290,7 @@ def delete_arc_agents(release_namespace, kube_config, kube_context, configuratio
 
 def helm_install_release(chart_path, subscription_id, kubernetes_distro, kubernetes_infra, resource_group_name, cluster_name,
                          location, onboarding_tenant_id, http_proxy, https_proxy, no_proxy, proxy_cert, private_key_pem,
-                         kube_config, kube_context, no_wait, values_file_provided, values_file, cloud_name, disable_auto_upgrade, enable_custom_locations, custom_locations_oid):
+                         kube_config, kube_context, no_wait, values_file_provided, values_file, cloud_name, disable_auto_upgrade, enable_custom_locations, custom_locations_oid, onboarding_timeout="5"):
```
Comment (alphaWizard) [inline src/connectedk8s/azext_connectedk8s/_utils.py:293]: Now its seconds, so 300 right?

Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/_utils.py:293]: its updated in the latest commit: https://github.com/Azure/azure-cli-extensions/pull/3711/commits/566b86f8fea332db6905b0bb605995485274775f

Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/_utils.py:293]: not sure why its not showing up here

**Review by alphaWizard — APPROVED** [review]

File: src/connectedk8s/azext_connectedk8s/_utils.py (hunk 4 above)
Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/_utils.py]: fixed

**Review by jsntcy — APPROVED** [review]

**Pull request merged**